### PR TITLE
chore: minor refactor to reduce copy/paste code in ScaledObject webhook

### LINKED ISSUE #5397: Refactor ScaledObject webhook
### Proposal

Minor refactoring, see details in https://github.com/kedacore/keda/pull/5377#discussion_r1455488231

### Use-Case

_No response_

### Is this a feature you are interested in implementing yourself?

No

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by wozniakjan**
all in the title :)
intending to merge after 2.13.0 release so keeping as draft for now

### Checklist

- [X] I have verified that my change is according to the [deprecations & breaking changes policy](https://github.com/kedacore/governance/blob/main/DEPRECATIONS.md)
- [X] Changelog has been updated and is aligned with our [changelog requirements](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#Changelog)
- [X] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))

Fixes #5397

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -63,6 +63,7 @@ Here is an overview of all new **experimental** features:
 ### Improvements
 
 - **General**: TODO ([#XXX](https://github.com/kedacore/keda/issues/XXX))
+- **General**: Minor refactor to reduce copy/paste code in ScaledObject webhook ([#5397](https://github.com/kedacore/keda/issues/5397))
```
Comment (zroubalik) [inline CHANGELOG.md:66]: Nit, we put non-user facing fixes to ### Other section

**Review by JorTurFer — APPROVED** [review]

**Pull request merged**